<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1415</width>
    <height>847</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>9</y>
      <width>541</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QPushButton" name="add_node">
           <property name="text">
            <string>Add Node</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="add_connection">
           <property name="text">
            <string>Add Conection</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLineEdit" name="f_node_for_connect"/>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout"/>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>TO</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="s_node_for_connect"/>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>weight:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="weight_le"/>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="delete_node">
           <property name="text">
            <string>Delete Node</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="reset_table">
           <property name="text">
            <string>Reset teble for</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="amount_node_for_table"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="set_from_table">
         <property name="text">
          <string>Set from table</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTableView" name="tableView"/>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>4</x>
      <y>11</y>
      <width>751</width>
      <height>671</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>670</y>
      <width>271</width>
      <height>91</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QPushButton" name="shiw_ways">
       <property name="text">
        <string>Show ways</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QPushButton" name="show_prev_way">
         <property name="text">
          <string>&lt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="show_next_way">
         <property name="text">
          <string>&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="deselect_all">
       <property name="text">
        <string>Deselect all</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>620</y>
      <width>261</width>
      <height>176</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Enter numbers vertices</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLineEdit" name="bfs_1"/>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>-&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="bfs_2"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="find_ways_bfs">
       <property name="text">
        <string>Find ways BFS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_way_dijkstra">
       <property name="text">
        <string>Find way Dijkstra</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_way_belman_ford">
       <property name="text">
        <string>Find way Belman-Ford</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_way_floyd_warshall">
       <property name="text">
        <string>Find way Floyd-Warshall</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>500</y>
      <width>192</width>
      <height>153</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5"/>
     </item>
     <item>
      <widget class="QPushButton" name="find_max_flow">
       <property name="text">
        <string>Find max flow</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="add_random_cost_edge">
       <property name="text">
        <string>Add random cost edges</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="delete_all_costs">
       <property name="text">
        <string>Delete costs</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_min_cost_max_flow">
       <property name="text">
        <string>Find min cost 2/3 max flow</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="change_orient">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>450</y>
      <width>391</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Дезориентировать/ориентировать граф</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>490</y>
      <width>236</width>
      <height>120</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_8">
     <item>
      <widget class="QPushButton" name="find_prima">
       <property name="text">
        <string>Алгоритм Прима</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_cruskala">
       <property name="text">
        <string>Алгоритм Краскала</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="code_prufer">
       <property name="text">
        <string>Код Прюфера</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_amount_ostov">
       <property name="text">
        <string>Количество остовных деревьев</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_6">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>700</y>
      <width>256</width>
      <height>89</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QPushButton" name="find_ailer_circle">
       <property name="text">
        <string>Найти эйлеров цикл</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="build_gamilton">
       <property name="text">
        <string>Достроить до Гамильтонова графа</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="find_comivoajor">
       <property name="text">
        <string>Задача комивояжёра</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1415</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
